SmartDeviceLink/SDLTouchManager.m

```diff
@@ -447,19 +448,22 @@ - (void)sdl_initializeSingleTapTimerAtPoint:(CGPoint)point {
         [self sdl_cancelSingleTapTimer];
     }
 
-    self.singleTapTimer = [NSTimer timerWithTimeInterval:self.tapTimeThreshold target:self selector:@selector(sdl_singleTapTimerCallback:) userInfo:@{@"point": [NSValue valueWithCGPoint:point]} repeats:NO];
-    [[NSRunLoop mainRunLoop] addTimer:self.singleTapTimer forMode:NSRunLoopCommonModes];
+    __weak typeof(self) weakSelf = self;
+    self.singleTapTimer = [[SDLTimer alloc] initWithDuration:self.tapTimeThreshold];
```

Comment:
If `self.tapTimeThreshold` is set to `0` then the `elapsedBlock` never gets called. This means that the second tap is always the second tap in a double tap regardless of the time between the taps.

Reply:
I changed SDLTimer to assert if a duration of `<= 0` is used

Reply:
I thought setting a duration of 0 disabled the double tap gesture? That is the current behavior, which is why I noticed that it stopped working. 

Should probably update the `tapTimeThreshold` documentation to state that `tapTimeThreshold` can't be less than 0.

Reply:
Disabling the double tap gesture with a `tapTimeThreshold` of 0 was never the case. From the `NSTimer` documentation:

> The number of seconds between firings of the timer. If ti is less than or equal to 0.0, this method chooses the nonnegative value of 0.1 milliseconds instead.

So it was just so small that you could never double tap in that time span. I updated the documentation.